### ISSUE TITLE
trader1.sqf

### ISSUE THREAD

**Issue by RAW5niper** [secondcoming/a3_exile_occupation#6]
I tried creating a new trader.sqf. Using M3Editor-Eden, added my objects, signs, etc. and added the trader's themselves. Exported it. Loaded it using a3_exile_occupation. Looks great...but...
The actual traders are not there. Everything is there except the physical traders.

I know it's not really an issue, or a problem. I'm just not sure what to do.

Any advice? Thanks in advance. :)

https://dl.dropboxusercontent.com/u/51067604/bashnya_trader.sqf

**Comment by secondcoming** [comment]
https://github.com/secondcoming/a3_exile_occupation/blob/development/source/a3_exile_occupation/trader/trader1.sqf

If you follow the example, place the small trader signs and the script automatically places the correct trader in front of each sign (only the small sign not the large ones)

If you look here you can see what it does:
https://github.com/secondcoming/a3_exile_occupation/blob/development/source/a3_exile_occupation/scripts/occupationTraders.sqf

**Comment by RAW5niper** [comment]
When I got this, I thought "great! simple fix, change the sign". Unfortunately, I changed the sign, and still
no trader. Also thought it maybe the way I edited (map). Put in props and signs, then clicked M3Editor in menu, and tried all (export objects, export all, export relative, etc), and pasted. None worked. This is a brilliant script, genius! If I can only figure out what I'm doing wrong.

**Comment by RAW5niper** [comment]
Very strange. Only shows up if I export objects (relative). Sign and trader show, but props only appear if I walk to a certain position (coords I put in config.cpp). And they (all) are not where I put them on the map; direction and position. Then I interacted with the trader (aircraft) but the purchased heli did not spawn on the helipad; spawned almost right along side the trader. I guess I'm getting closer. Exporting objects relative...relative to what? What center?

**Comment by secondcoming** [comment]
I used the m3editor plugin for eden to generate the file, try using that

**Comment by RAW5niper** [comment]
Yeah. That's what I'm using too.

**Comment by secondcoming** [comment]
did you do the copy to clipboard export?

**Comment by RAW5niper** [comment]
Yes.
M3Editor-->Export Objects (relative)--->copy

That gave me this:
private _objects = [
    ["Exile_Sign_Aircraft",[23.5774,8.77441,0],183.733,[false,false]],
    ["Land_BarGate_F",[-80.8164,-28.7344,0],287.806,[false,false]],
    ["CUP_A2_tent_west",[19.0271,11.7715,0.00259399],270.435,[false,false]],
    ["Land_HelipadSquare_F",[39.856,10.915,0],1.16551,[false,false]]
];

private _center = [0,0,0];
{
    private _object = (_x select 0) createVehicle [0,0,0];
    _object setDir (_x select 2);
    _object setPosATL (_center vectorAdd (_x select 1));
    _object enableSimulationGlobal ((_x select 3) select 0);
    _object allowDamage ((_x select 3) select 1);
} forEach _objects;

But I noticed your original trader1.sqf has this line above private_objects:
private _traderPosition = _this select 0;

And these lines after:
private _center = _traderPosition;
{
    private _object = createVehicle [(_x select 0), [0,0,0], [], 0, "CAN_COLLIDE"];
    _object setDir (_x select 2);
    _object setPos (_center vectorAdd (_x select 1));
    _object enableSimulationGlobal false; 
    _object allowDamage false;
    //private _newPos = getPos _object;
    //_object setPos _newPos;

} forEach _objects;

Which is different from the one I got when I copy and paste.

Map I'm using is Chernarus, which has really crappy traders from the host provider.
I can do an awesome job at creating traders with this script, once I get it to work.

**Comment by secondcoming** [comment]
yes, you need to use the rest of the code from the example, just replace the objects within the square brackets.

**Comment by RAW5niper** [comment]
Thought so. Getting closer. :) Why would my purchase not have spawned on the helipad I put down? What's the trick to that? I suspect I'll have the same issue with vehicle trader.

**Comment by secondcoming** [comment]
if you use the override to place the purchased vehicles on helipads then
check you are placing the correct one, there are several different types

**Comment by RAW5niper** [comment]
Couldn't figure out why objects and traders were not where I placed them (all moved around). So I created the city (objects) using a3_custom. And added the traders using the mission initPlayerLocal.sqf. But for some reason, some of the props (objects) don't show up (only few); they do on some reboots though. Maybe there's a limitation. Thanks for your help secondcoming! Can't wait for exilemod site to be back up.
